public/app/features/dashboard-scene/serialization/transformToV1TypesUtils.ts

```diff
@@ -0,0 +1,131 @@
+  VariableSort as VariableSortV1,
+  DashboardCursorSync as DashboardCursorSyncV1,
+  defaultDashboardCursorSync,
+} from '@grafana/schema';
+import {
+  DashboardCursorSync,
+  MappingType,
+  VariableHide,
+  VariableRefresh,
+  VariableSort,
+  FieldConfigSource,
+  SpecialValueMatch,
+} from '@grafana/schema/dist/esm/schema/dashboard/v2alpha0/dashboard.gen';
+
+export function transformVariableRefreshToEnumV1(refresh?: VariableRefresh): VariableRefreshV1 {
+  switch (refresh) {
+    case VariableRefresh.Never:
+      return VariableRefreshV1.never;
+    case VariableRefresh.OnDashboardLoad:
+      return VariableRefreshV1.onDashboardLoad;
+    case VariableRefresh.OnTimeRangeChanged:
+      return VariableRefreshV1.onTimeRangeChanged;
+    default:
+      return VariableRefreshV1.never;
+  }
+}
+
+export function transformVariableHideToEnumV1(hide?: VariableHide): VariableHideV1 {
+  switch (hide) {
+    case VariableHide.DontHide:
+      return VariableHideV1.dontHide;
+    case VariableHide.HideLabel:
+      return VariableHideV1.hideLabel;
+    case VariableHide.HideVariable:
+      return VariableHideV1.hideVariable;
+    default:
+      return VariableHideV1.dontHide;
+  }
+}
+
+export function transformSortVariableToEnumV1(sort?: VariableSort): VariableSortV1 {
+  switch (sort) {
+    case VariableSort.Disabled:
+      return VariableSortV1.disabled;
+    case VariableSort.NumericalAsc:
+      return VariableSortV1.numericalAsc;
+    case VariableSort.NumericalDesc:
+      return VariableSortV1.numericalDesc;
+    case VariableSort.AlphabeticalAsc:
+      return VariableSortV1.alphabeticalAsc;
+    case VariableSort.AlphabeticalDesc:
+      return VariableSortV1.alphabeticalDesc;
+    default:
+      return VariableSortV1.disabled;
+  }
+}
+
+export function transformCursorSyncV2ToV1(cursorSync: DashboardCursorSync): DashboardCursorSyncV1 {
+  switch (cursorSync) {
+    case DashboardCursorSync.Crosshair:
+      return DashboardCursorSyncV1.Crosshair;
+    case DashboardCursorSync.Tooltip:
+      return DashboardCursorSyncV1.Tooltip;
+    case DashboardCursorSync.Off:
+      return DashboardCursorSyncV1.Off;
+    default:
+      return defaultDashboardCursorSync;
+  }
+}
+
+function transformSpecialValueMatchToV1(match: SpecialValueMatch): SpecialValueMatchV1 {
+  switch (match) {
+    case SpecialValueMatch.True:
+      return SpecialValueMatchV1.True;
+    case SpecialValueMatch.False:
+      return SpecialValueMatchV1.False;
+    case SpecialValueMatch.Null:
+      return SpecialValueMatchV1.Null;
+    case SpecialValueMatch.NotANumber:
+      return SpecialValueMatchV1.NaN;
+    case SpecialValueMatch.NullAndNaN:
+      return SpecialValueMatchV1.NullAndNaN;
+    case SpecialValueMatch.Empty:
+      return SpecialValueMatchV1.Empty;
+    default:
+      throw new Error(`Unknown match type: ${match}`);
+  }
+}
+
+export function transformValueMappingsToV1(fieldConfig: FieldConfigSource): FieldConfigSourceV1 {
```

Comment:
This function wouldn't be necessary if we could do this in CUE:
```
#ValueMapping: ValueMap | RangeMap | RegexMap | SpecialValueMap

ValueMap: {
  type: #ValueMapping.ValueMap
  options: [string]: ValueMappingResult
}
```

We are currently doing this: 
```
ValueMap: {
  type: MappingType & "value"
} 
```
which results in:
```
export interface ValueMap {
	type: "value";
	// Map with <value_to_match>: ValueMappingResult. For example: { "10": { text: "Perfection!", color: "green" } }
	options: Record<string, ValueMappingResult>;
}
```
This causes a mismatch with what is expected, which is this shape:
https://github.com/grafana/grafana/blob/d523fb46b2a047f61ea40e7cb92a7a4fa9c2a026/packages/grafana-data/src/types/valueMapping.ts#L32-L34

This is because in TypeScript, enums with different member names are considered different types, even if the values are the same. This means that we cannot interchange these enums without causing a type error.

Reply:
All of this is tech debt we should fix at some point 😢 Hopefully, a major release of Scenes with all these new types used, will help us to solve this translation.